gadget/mountedfilesystem.go

```diff
@@ -197,9 +195,13 @@ func (m *MountedFilesystemWriter) observedWriteFileOrSymlink(volumeRoot, src, ds
 		// with content in this file
 		After: src,
 	}
-	if err := observe(m.observer, ContentWrite, m.ps, volumeRoot, dst, data); err != nil {
+	act, err := observe(m.observer, ContentWrite, m.ps, volumeRoot, dst, data)
+	if err != nil {
 		return fmt.Errorf("cannot observe file write: %v", err)
 	}
+	if act == ChangePreserveBefore && osutil.FileExists(dst) {
```

Comment:
is this the thing we discussed? where instead maybe we should change s/PreserveBefore/Ignore/ and just return even if the file doesn't exist?

Reply:
This is done now. I've added an additional stamp for files from changes that the observer asked to be ignored.